```{r}
rm(list=ls())
```

```{r}
library(reshape2)
library(ggpubr)
library(knitr)
library(DESeq2)
library(ggplot2)
library(dplyr)
library(ggrepel)
library(ggalt)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(ReactomePA)
library(tidyverse)
library(enrichplot)
library(pheatmap)
library(clusterProfiler)
library(openxlsx)
library(limma)
library(GSVA)
library(msigdbr)
library(BiocParallel)
library(ggh4x)
library(scales)
```

```{r}
subread.counts <- readRDS("subreadcounts_malt1mutants_notraf6ko.rds")
```

```{r}
rawtable <- read.csv("rawtable_malt1mutants_notraf6ko.csv", sep = ";")
colnames(rawtable) <- sub("X", "", colnames(rawtable))
rcounts <- rawtable[,-1]
rownames(rcounts) <- rawtable[,1]
```

```{r}
# subreadcounts_stat <- subread.counts[["stat"]]
# 
# cols_to_keep <- c("Status", intersect(colnames(subreadcounts_stat), colnames(rcounts)))
# subreadcounts_stat_subset <- subreadcounts_stat[, cols_to_keep]
# 
# subread.counts[["stat"]] <- subreadcounts_stat_subset
# 
# subread.counts[["targets"]] <- subset(subread.counts[["targets"]], subread.counts[["targets"]] %in% cols_to_keep)
# 
# subreadcounts_counts <- subread.counts[["counts"]]
# 
# cols_to_keep <- c(intersect(colnames(subreadcounts_stat), colnames(rcounts)))
# subreadcounts_counts_subset <- subreadcounts_counts[, cols_to_keep]
# 
# subread.counts[["counts"]] <- subreadcounts_counts_subset
# 
# saveRDS(subread.counts, "subreadcounts_malt1mutants_notraf6ko.rds")
```


```{r}
cnt <- rcounts[(rowSums(rcounts)>0),] 
dim(cnt)
head(cnt)
```

```{r}
cnt <- rcounts[(rowSums(rcounts)>0),] 
colnames(cnt) <- sub("Aligned.out.bam", "", colnames(cnt))
colnames(cnt) <- sub("23L00", "S", colnames(cnt))
dim(cnt)
head(cnt)
```

```{r}
human_biotype <- read.csv("/Users/goeksu.avar/Desktop/Krappmann_2/Genomes/GCF_000001405.39_GRCh38.p13_feature_table.tsv", row.names = NULL, sep="\t")
```

```{r}
coding_genes <- subset(human_biotype,  class == "protein_coding")
```

```{r}
cnt <- subset(cnt, rownames(cnt) %in% coding_genes$symbol)
```

```{r}
geneLength <- subread.counts$annotation %>%
  dplyr::select("GeneID", "Length")
```

```{r}
#make countdata a dataframe
countdata <- data.frame(GeneID=rownames(cnt), cnt)
countdata <- geneLength %>%
  inner_join(countdata) %>%
  mutate(Length=ifelse(Length-151+1>0,Length-151+1,1))

tpm<-countdata %>%
  pivot_longer( cols = starts_with("S"), names_to = "sample",
                values_to = "cnt") %>%
  group_by(sample) %>%
  mutate(rpk = cnt/ (Length/1000) , libSize = sum(rpk), tpm = rpk / libSize *1e6 )
```

```{r}
tpm <- tpm %>% mutate( minTPM= min(tpm[tpm>0]), log10TPM = log10(tpm+minTPM))
head(tpm)
```

```{r}
#let's remove columns with intermediate results
tpm <- tpm %>% dplyr::select(-libSize,-rpk,-minTPM)

png(file="TPM_filtered.png")
ggplot(tpm,aes(x=sample,y=log10TPM))+
  geom_boxplot()+
  coord_flip()+
  ylab(expression(log[10](TPM)))+
  ggtitle("TPM (Filtered)")
dev.off()

openxlsx::write.xlsx(tpm, "TPM.xlsx")
```

```{r}
tpm %>% summarise(no.of.genes=sum( tpm>=1 ))
```

```{r}
#the tibble tpm is grouped according to samples, this grouping needs to be removed
tpm2<-tpm %>%
  ungroup() %>%
  group_by(GeneID) %>%
  dplyr::filter(sum(tpm<1) < 10)

cnt.tib <- tpm2 %>% dplyr::select(GeneID,sample,cnt) %>% pivot_wider(names_from = sample,
values_from = cnt)
#DEseq wants a count matrix not a tibble
cnt.filt<-as.matrix(cnt.tib[,-1])
rownames(cnt.filt)<-cnt.tib$GeneID
```

```{r}
coldata <- read.csv("/Users/goeksu.avar/Desktop/Krappmann_2/MALT1Mutants/coldata_malt1mutants_notraf6ko.csv", sep = ";")

coldata[coldata$Sample_ID == "Sample_23L001558", "Sample_ID"] <- "Sample_23L001546"
coldata[coldata$Sample_Name == "S1558", "Sample_Name"] <- "S1546"

#coldata <- subset(coldata, !Sample_Name %in% c("S1546", "S1558"))
coldata <- as.matrix(coldata, rownames.force = NA)
rownames(coldata) <- coldata[,2]
coldata <- coldata[,-1]
```

```{r}
#cnt <- dplyr::select(cnt, -c("S1546", "S1558"))
```

```{r}
# Rename the "S1558" column to "S1546"
colnames(cnt)[colnames(cnt) == "S1558"] <- "S1546"
colnames(cnt.filt)[colnames(cnt.filt) == "S1558"] <- "S1546"
```

```{r}
all(rownames(coldata) %in% colnames(cnt.filt))
```

```{r}
all(rownames(coldata) == colnames(cnt.filt))
```

```{r}
dds <- DESeqDataSetFromMatrix(countData = cnt.filt,
                              colData = coldata,
                              design = ~Batch+Sample_Group)
dds
```

```{r}
resultsNames(dds)
```

```{r}
dds<- DESeq(dds)
saveRDS(dds, file = "dds_malt1mutants.RDS")
write.csv(dds@assays@data@listData[["counts"]], "DESeq2_Counts_malt1mutants.csv")
```

```{r}
vsd <- vst(dds, blind=FALSE)
vsdMat <- assay(vsd)
```

```{r}
vsdmelted <- melt(vsdMat)

png("VSD.png")
ggplot(vsdmelted, aes(x=Var2, y=value))+
  geom_boxplot()+
  coord_flip()+
  ylab("Variance stabilized counts")+
  xlab("Sample")+
  ggtitle("VST")
dev.off()
```

```{r}
pcaData <- plotPCA(vsd, intgroup=c("Sample_Group"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
pcaData$Sample_Group <- factor(pcaData$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))


pdf(file="PCA_vst.pdf", height=4, width=4)
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=7)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 10))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 10))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line())+
  geom_text_repel(aes(label=name), size=3.5, color="black", max.overlaps = Inf)+
  coord_fixed()+
  scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                              "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                              "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                              "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                              "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998"))+
  # geom_encircle(data=subset(pcaData, Sample_Group=="BJAB_MALT_KO_C11_L232LI_Mock"), fill="#1B9CD8", color = "#1B9CD8", alpha = 0.25, expand = 0.3, spread = 0.5, s_shape = 1)+
  # geom_encircle(data=subset(pcaData, Sample_Group=="BJAB_MALT_KO_C11_L232LI_WT"), fill="#F39205", color = "#F39205", alpha = 0.25, s_shape = 0.6)+
  # geom_encircle(data=subset(pcaData, Sample_Group=="BJAB_MALT_KO_C11_L232LI_CA"), fill="#95C120", color = "#95C120", alpha = 0.25, s_shape = 0.6)+
  # geom_encircle(data=subset(pcaData, Sample_Group=="BJAB_MALT_KO_C11_L232LI_EA"), fill="#016633", color = "#016633", alpha = 0.25, s_shape = 0.6)+
  # geom_encircle(data=subset(pcaData, Sample_Group=="BJAB_MALT_KO_C11_L232LI_CA_EA"), fill="#CA9998", color = "#CA9998", alpha = 0.25, s_shape = 0.6)+
  guides(color=FALSE)
dev.off()

pcaData <- plotPCA(vsd, intgroup=c("Batch"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

pdf(file="PCA_vst_BATCHES.pdf", height=8, width=10)
ggplot(pcaData, aes(PC1, PC2, color=Batch)) +
  geom_point(size=7, shape=21, fill="white", stroke=2)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 27))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 27))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line())+
  geom_text_repel(aes(label=name), size=5, color="black", max.overlaps = Inf)+
  coord_fixed()
dev.off()
```

```{r}
mm <- model.matrix(~Sample_Group, colData(vsd))
vsdMat <- limma::removeBatchEffect(vsdMat, batch=vsd$Batch, design=mm)
assay(vsd) <- vsdMat
```

```{r}
pcaData <- plotPCA(vsd, intgroup=c("Sample_Group"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
# pcaData$Sample_Group <- factor(pcaData$Sample_Group, levels = c("BJAB_API2_MALT1_Mock",
#                                                                 "BJAB_API2_MALT1_WT",
#                                                                 "BJAB_API2_MALT1_CA",
#                                                                 "BJAB_API2_MALT1_EA"))

pdf(file="PCA_vst_batchremoved.pdf", height=8, width=10)
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=7, shape=21, fill="white", stroke=2)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 27))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme(axis.title = element_text(size = 27))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line())+
  geom_text_repel(aes(label=name), size=5, color="black", max.overlaps = Inf)+
  coord_fixed()
dev.off()

pcaData <- plotPCA(vsd, intgroup=c("Batch"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

pdf(file="PCA_vst_batchremoved_BATCHES.pdf", height=8, width=10)
ggplot(pcaData, aes(PC1, PC2, color=Batch)) +
  geom_point(size=7, shape=21, fill="white", stroke=2)+
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 27))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) +
  theme(axis.title = element_text(size = 27))+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line())+
  geom_text_repel(aes(label=name), size=5, color="black", max.overlaps = Inf)+
  coord_fixed()
dev.off()
```

```{r}
res_bjab_wt_mock <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_MALT_KO_C11_L232LI_WT","BJAB_MALT_KO_C11_L232LI_Mock"), alpha = 0.05)
summary(res_bjab_wt_mock)

res_bjab_wt_mock_list<-data.frame(gene = rownames(res_bjab_wt_mock), res_bjab_wt_mock,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 

res_bjab_wt_mock_list$padj[res_bjab_wt_mock_list$padj == 0] <- .Machine$double.xmin

res_bjab_wt_mock_list <- res_bjab_wt_mock_list %>% 
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
  rownames_to_column(var = "gene") %>% 
  arrange(desc(abs(LFCminuslogp)))



res_bjab_ca_wt <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_MALT_KO_C11_L232LI_CA","BJAB_MALT_KO_C11_L232LI_WT"), alpha = 0.05)
summary(res_bjab_ca_wt)

res_bjab_ca_wt_list<-data.frame(gene = rownames(res_bjab_ca_wt), res_bjab_ca_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 

res_bjab_ca_wt_list$padj[res_bjab_ca_wt_list$padj == 0] <- .Machine$double.xmin

res_bjab_ca_wt_list <- res_bjab_ca_wt_list %>% 
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
  rownames_to_column(var = "gene") %>% 
  arrange(desc(abs(LFCminuslogp)))



res_bjab_ea_wt <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_MALT_KO_C11_L232LI_EA","BJAB_MALT_KO_C11_L232LI_WT"), alpha = 0.05)
summary(res_bjab_ea_wt)

res_bjab_ea_wt_list<-data.frame(gene = rownames(res_bjab_ea_wt), res_bjab_ea_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 

res_bjab_ea_wt_list$padj[res_bjab_ea_wt_list$padj == 0] <- .Machine$double.xmin

res_bjab_ea_wt_list <- res_bjab_ea_wt_list %>% 
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
  rownames_to_column(var = "gene") %>% 
  arrange(desc(abs(LFCminuslogp)))


res_bjab_caea_wt <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_MALT_KO_C11_L232LI_CA_EA","BJAB_MALT_KO_C11_L232LI_WT"), alpha = 0.05)
summary(res_bjab_ea_wt)

res_bjab_caea_wt_list<-data.frame(gene = rownames(res_bjab_caea_wt), res_bjab_caea_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 

res_bjab_caea_wt_list$padj[res_bjab_caea_wt_list$padj == 0] <- .Machine$double.xmin

res_bjab_caea_wt_list <- res_bjab_caea_wt_list %>% 
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
  rownames_to_column(var = "gene") %>% 
  arrange(desc(abs(LFCminuslogp)))


res_bjab_ca_vs_ea <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_MALT_KO_C11_L232LI_CA","BJAB_MALT_KO_C11_L232LI_EA"), alpha = 0.05)
summary(res_bjab_ca_vs_ea)

res_bjab_ca_vs_ea_list<-data.frame(gene = rownames(res_bjab_ca_vs_ea), res_bjab_ca_vs_ea,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 

res_bjab_ca_vs_ea_list$padj[res_bjab_ca_vs_ea_list$padj == 0] <- .Machine$double.xmin

res_bjab_ca_vs_ea_list <- res_bjab_ca_vs_ea_list %>% 
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
  rownames_to_column(var = "gene") %>% 
  arrange(desc(abs(LFCminuslogp)))
```

```{r}
# res_t6ko_wt_mock <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_TRAF6_KO_MALT_KO_C11_L232LI_WT","BJAB_TRAF6_KO_MALT_KO_C11_L232LI_Mock"), alpha = 0.1)
# summary(res_t6ko_wt_mock)
# 
# res_t6ko_wt_mock_list<-data.frame(gene = rownames(res_t6ko_wt_mock), res_t6ko_wt_mock,
# stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 
# 
# res_t6ko_wt_mock_list$padj[res_t6ko_wt_mock_list$padj == 0] <- .Machine$double.xmin
# 
# res_t6ko_wt_mock_list <- res_t6ko_wt_mock_list %>% 
#   dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
#   rownames_to_column(var = "gene") %>% 
#   arrange(desc(abs(LFCminuslogp)))
# 
# 
# 
# 
# res_t6ko_ca_wt <- DESeq2::results(dds, contrast=c("Sample_Group","BJAB_TRAF6_KO_MALT_KO_C11_L232LI_CA","BJAB_TRAF6_KO_MALT_KO_C11_L232LI_WT"), alpha = 0.1)
# summary(res_t6ko_ca_wt)
# 
# res_t6ko_ca_wt_list<-data.frame(gene = rownames(res_t6ko_ca_wt), res_t6ko_ca_wt,
# stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") 
# 
# res_t6ko_ca_wt_list$padj[res_t6ko_ca_wt_list$padj == 0] <- .Machine$double.xmin
# 
# res_t6ko_ca_wt_list <- res_t6ko_ca_wt_list %>% 
#   dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% 
#   rownames_to_column(var = "gene") %>% 
#   arrange(desc(abs(LFCminuslogp)))
```

```{r}
reactome_genesets <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME") %>% 
  dplyr::select(gs_name, gene_symbol)

reactome_list <- unlist(split(reactome_genesets, f = reactome_genesets$gs_name), recursive = F)

gsva_reactome <- as.data.frame(gsva(as.matrix(vsdMat), reactome_list, verbose=FALSE))
rownames(gsva_reactome) <- sub(".gene_symbol", "", rownames(gsva_reactome))
gsva_reactome$Pathway <- rownames(gsva_reactome)
```

```{r}
oncogenic_genesets <- msigdbr(species = "Homo sapiens", category = "C6") %>% 
  dplyr::select(gs_name, gene_symbol)

oncogenic_list <- unlist(split(oncogenic_genesets, f = oncogenic_genesets$gs_name), recursive = F)

gsva_oncogenic <- as.data.frame(gsva(as.matrix(vsdMat), oncogenic_list, verbose=FALSE))
rownames(gsva_oncogenic) <- sub(".gene_symbol", "", rownames(gsva_oncogenic))
gsva_oncogenic$Pathway <- rownames(gsva_oncogenic)
```


```{r}
tftarget_genesets <- msigdbr(species = "Homo sapiens", category = "C3") %>% 
  dplyr::select(gs_name, gene_symbol)

tftarget_list <- unlist(split(tftarget_genesets, f = tftarget_genesets$gs_name), recursive = F)

gsva_tftarget <- as.data.frame(gsva(as.matrix(vsdMat), tftarget_list, verbose=FALSE))
rownames(gsva_tftarget) <- sub(".gene_symbol", "", rownames(gsva_tftarget))
gsva_tftarget$Pathway <- rownames(gsva_tftarget)
```

```{r}
gsva_tftarget_temp <- as.data.frame(t(dplyr::select(gsva_tftarget, -Pathway)))
gsva_tftarget_temp$Sample_Name <- rownames(gsva_tftarget_temp)
gsva_tftarget_annotated <- merge(gsva_tftarget_temp, coldata, by =  "Sample_Name")
```

```{r}
ggplot(gsva_tftarget_annotated, aes(x=Sample_Group, y=NFKB_C))+
  geom_boxplot()+
  geom_jitter(width = 0.1)
```

```{r}
signature_database <- read.csv("/Users/goeksu.avar/Desktop/Krappmann_2/SignatureDB_030920.tsv", sep = "\t")
```

```{r}
signatures <- signature_database %>%
    dplyr::select(X____Short.signature.name, X____Gene.ID.concensus, X____Gene.symbol.concensus) %>%
    tidyr::unnest(cols = c()) %>%
    dplyr::rename("gs_name" = X____Short.signature.name, 
                  "entrez_gene" = X____Gene.ID.concensus, 
                  "gene_symbol" = X____Gene.symbol.concensus)
```

```{r}
signatures_nfkb <- subset(signatures, gs_name %in% c("NFkB-2", "NFkB-9", "NFkB-10"))
signatures_nfkb_list <- unlist(split(signatures_nfkb, f = signatures_nfkb$gs_name), recursive = F)
```

```{r}
gsva_signatures_nfkb <- as.data.frame(gsva(as.matrix(vsdMat), signatures_nfkb_list, verbose=FALSE), method = "gsva")
rownames(gsva_signatures_nfkb) <- sub(".gene_symbol", "", rownames(gsva_signatures_nfkb))
gsva_signatures_nfkb$Pathway <- rownames(gsva_signatures_nfkb)
```

```{r}
gsva_signatures_nfkb_temp <- as.data.frame(t(dplyr::select(gsva_signatures_nfkb, -Pathway)))
gsva_signatures_nfkb_temp$Sample_Name <- rownames(gsva_signatures_nfkb_temp)
gsva_signatures_nfkb_annotated <- merge(gsva_signatures_nfkb_temp, coldata, by = "Sample_Name")
```

```{r}
gsva_signatures_nfkb_annotated$Sample_Group <- factor(gsva_signatures_nfkb_annotated$Sample_Group, 
                                                      levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                 "BJAB_MALT_KO_C11_L232LI_WT",
                                                                 "BJAB_MALT_KO_C11_L232LI_CA",
                                                                 "BJAB_MALT_KO_C11_L232LI_EA",
                                                                 "BJAB_MALT_KO_C11_L232LI_CA_EA"))

gsva_signatures_nfkb_annotated_melted <- reshape2::melt(gsva_signatures_nfkb_annotated,
                                                        variable.name = "geneset",
                                                        value.name = "gsva_score")

gsva_signatures_nfkb_annotated_melted$geneset <- factor(gsva_signatures_nfkb_annotated_melted$geneset, 
                                                      levels = c("NFkB-2",
                                                                 "NFkB-9",
                                                                 "NFkB-10"))

my_comparisons <- list(c("BJAB_MALT_KO_C11_L232LI_WT", "BJAB_MALT_KO_C11_L232LI_Mock"), 
                       c("BJAB_MALT_KO_C11_L232LI_CA", "BJAB_MALT_KO_C11_L232LI_WT"), 
                       c("BJAB_MALT_KO_C11_L232LI_EA", "BJAB_MALT_KO_C11_L232LI_WT"), 
                       c("BJAB_MALT_KO_C11_L232LI_CA_EA", "BJAB_MALT_KO_C11_L232LI_WT"))

pdf("gsva_scores_malt1mutants.pdf", width = 5, height = 5)
ggplot(gsva_signatures_nfkb_annotated_melted, aes(x=Sample_Group, y=gsva_score, color=Sample_Group)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(width = 0.1) +
  #stat_summary(fun = mean, geom = "point", shape = 20, size = 3, color = "red", show.legend = FALSE) +
  theme_bw()+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),
        plot.background = element_blank(), 
        axis.line = element_line(), 
        axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
  facet_wrap(~geneset, ncol = 3, nrow = 1)+
  scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                              "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                              "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                              "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                              "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998"))+
  guides(color=FALSE)
dev.off()
```

```{r}
#Regulatory targets
C3_t2g <- msigdbr(species = "Homo sapiens", category = "C3") %>% dplyr::select(gs_name, entrez_gene)
```

```{r}
# set the number of parallel workers to 5
register(MulticoreParam(workers = 5))


#Define the GSEA function

enrichment_analysis <- function(group1, group2) {
  res <- DESeq2::results(dds, contrast=c("Sample_Group", group1, group2), alpha = 0.05)
  res <- data.frame(rn=rownames(res), res) %>% na.omit()

  gsea <- res$padj 
  names(gsea) <- res$rn
  ensgEntrez_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(gsea), columns='ENTREZID', keytype='SYMBOL')
  df_gsea <- ensgEntrez_gsea %>%
    dplyr::inner_join(res, by=c("SYMBOL"="rn")) %>% 
    dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
    dplyr::group_by(ENTREZID) %>%
    dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj)) 
  lfc <- df_gsea$lfc 
  names(lfc) <- df_gsea$ENTREZID

  SignatureDB <- GSEA(sort((lfc), decreasing=T), TERM2GENE = signatures, eps = 0, pvalueCutoff = 1)
  SignatureDB <- setReadable(SignatureDB, 'org.Hs.eg.db', 'ENTREZID')

  Reactome <- gsePathway(sort((lfc), decreasing=T), organism = "human", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps=0)
  Reactome <- setReadable(Reactome, 'org.Hs.eg.db', 'ENTREZID')

  MSigDB <- GSEA(sort((lfc), decreasing=T), pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, TERM2GENE = C3_t2g, eps = 0)
  MSigDB <- setReadable(MSigDB, 'org.Hs.eg.db', 'ENTREZID')

    # check for NAs in setReadable output and remove any rows with NAs
  if (any(is.na(SignatureDB))) {
    SignatureDB <- SignatureDB[!is.na(SignatureDB),]
  }
  if (any(is.na(Reactome))) {
    Reactome <- Reactome[!is.na(Reactome),]
  }
  if (any(is.na(MSigDB))) {
    MSigDB <- MSigDB[!is.na(MSigDB),]
  }

  return(list(SignatureDB=SignatureDB, Reactome=Reactome, MSigDB=MSigDB))
}

group_comparisons <- list(
  c("BJAB_MALT_KO_C11_L232LI_WT", "BJAB_MALT_KO_C11_L232LI_Mock"),
  c("BJAB_MALT_KO_C11_L232LI_CA", "BJAB_MALT_KO_C11_L232LI_WT"),
  c("BJAB_MALT_KO_C11_L232LI_EA", "BJAB_MALT_KO_C11_L232LI_WT"),
  c("BJAB_MALT_KO_C11_L232LI_CA_EA", "BJAB_MALT_KO_C11_L232LI_WT")
)

# execute the function for each group comparison
gsea_results <- list()

for (i in seq_along(group_comparisons)) {
  comparison <- group_comparisons[[i]]
  group1 <- comparison[[1]]
  group2 <- comparison[[2]]
  
  res <- enrichment_analysis(group1, group2)
  
  gsea_results[[i]] <- list(
    group1 = group1,
    group2 = group2,
    SignatureDB = res$SignatureDB,
    Reactome = res$Reactome,
    MSigDB = res$MSigDB
  )
}
```
```{r}
# sort the GSEA outputs by NES in decreasing order and save them as xlsx files
for (i in seq_along(gsea_results)) {
  group1 <- gsea_results[[i]]$group1
  group2 <- gsea_results[[i]]$group2
  
  # sort SignatureDB by NES
  SignatureDB <- gsea_results[[i]][["SignatureDB"]]@result
  SignatureDB <- SignatureDB[order(-SignatureDB$NES), ]
  write.xlsx(SignatureDB, file = paste0(group1, "_vs_", group2, "_SignatureDB_gsea.xlsx"), rowNames = FALSE)

  # sort Reactome by NES
  Reactome <- gsea_results[[i]][["Reactome"]]@result
  Reactome <- Reactome[order(-Reactome$NES), ]
  write.xlsx(Reactome, file = paste0(group1, "_vs_", group2, "_Reactome_gsea.xlsx"), rowNames = FALSE)

  # sort MSigDB by NES
  MSigDB <- gsea_results[[i]][["MSigDB"]]@result
  MSigDB <- MSigDB[order(-MSigDB$NES), ]
  write.xlsx(MSigDB, file = paste0(group1, "_vs_", group2, "_MSigDB_gsea.xlsx"), rowNames = FALSE)
}
```

```{r}
pdf(file="Volcano_wt_mock_malt1ko.pdf", width=24, height=21)
ggplot(res_bjab_wt_mock_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(res_bjab_wt_mock_list, padj<=0.05), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(res_bjab_wt_mock_list, padj<0.05 & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(res_bjab_wt_mock_list, padj>0.05 & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.05), linetype="dashed")+
  geom_label_repel(data=subset(res_bjab_wt_mock_list, padj<=0.05 & abs(log2FoldChange)>1), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
pdf(file="Volcano_ca_wt_malt1ko.pdf", width=24, height=21)
ggplot(res_bjab_ca_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(res_bjab_ca_wt_list, padj<=0.05), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(res_bjab_ca_wt_list, padj<0.05 & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(res_bjab_ca_wt_list, padj>0.05 & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.05), linetype="dashed")+
  geom_label_repel(data=subset(res_bjab_ca_wt_list, padj<=0.05 & abs(log2FoldChange)>1), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```
```{r}
pdf(file="Volcano_ea_wt_malt1ko.pdf", width=24, height=21)
ggplot(res_bjab_ea_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(res_bjab_ea_wt_list, padj<=0.05), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(res_bjab_ea_wt_list, padj<0.05 & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(res_bjab_ea_wt_list, padj>0.05 & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.05), linetype="dashed")+
  geom_label_repel(data=subset(res_bjab_ea_wt_list, padj<=0.05), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
pdf(file="Volcano_caea_wt_malt1ko.pdf", width=24, height=21)
ggplot(res_bjab_caea_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(res_bjab_caea_wt_list, padj<=0.05), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(res_bjab_caea_wt_list, padj<0.05 & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(res_bjab_caea_wt_list, padj>0.05 & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.05), linetype="dashed")+
  geom_label_repel(data=subset(res_bjab_caea_wt_list, padj<=0.05), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
# pdf(file="Volcano_t6_ca_wt_malt1ko.pdf", width=24, height=21)
# ggplot(res_t6ko_ca_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
#   geom_point(size=2.25,color="grey", alpha=0.5)+
#   geom_point(data=subset(res_t6ko_ca_wt_list, padj<=0.05), size=2.25,color="darkorange1", alpha=1)+
#   geom_point(data=subset(res_t6ko_ca_wt_list, padj<0.05 & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
#   geom_point(data=subset(res_t6ko_ca_wt_list, padj>0.05 & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
#   theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
#         panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
#   xlab("Log2FoldChange")+
#   theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
#   ylab("-log10(p-adj)")+
#   theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
#   geom_vline(xintercept = 1, linetype="dashed")+
#   geom_vline(xintercept = -1, linetype="dashed")+
#   geom_hline(yintercept = -log10(0.05), linetype="dashed")+
#   geom_label_repel(data=subset(res_t6ko_ca_wt_list, padj<=0.05), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
# dev.off()
```

```{r}
significant_genes_wt_mock_UP <- subset(res_bjab_wt_mock_list, padj <= 0.01 & log2FoldChange >= 1.3) %>%
  dplyr::arrange(padj)

significant_genes_ca_wt_DN <- subset(res_bjab_ca_wt_list, padj <= 0.01 & log2FoldChange <= -0.5)  %>%
  dplyr::arrange(padj)

significant_genes_ea_wt_DN <- subset(res_bjab_ea_wt_list, padj <= 0.01 & log2FoldChange <= -0.5)  %>%
  dplyr::arrange(padj)

significant_genes_caea_wt_DN <- subset(res_bjab_caea_wt_list, padj <= 0.01 & log2FoldChange <= -0.5)  %>%
  dplyr::arrange(padj)

significant_genes_ca_vs_ea <- subset(res_bjab_ca_vs_ea_list, padj <= 0.01 & abs(log2FoldChange) >= 0.5) %>%
  dplyr::arrange(padj)
```

```{r}
ca_dependent_genes_list <- intersect(significant_genes_wt_mock_UP$gene, significant_genes_ca_wt_DN$gene)
ca_dependent_genes_list <- setdiff(ca_dependent_genes_list, significant_genes_ea_wt_DN$gene)
ca_dependent_genes_list <- intersect(ca_dependent_genes_list, significant_genes_ca_vs_ea$gene)
print(ca_dependent_genes_list)

ordered_protease_dependent_genes <- subset(res_bjab_ca_wt_list, gene %in% ca_dependent_genes_list) %>%
  dplyr::arrange(padj)

print(ordered_protease_dependent_genes$gene)
```

```{r}
ea_dependent_genes_list <- intersect(significant_genes_wt_mock_UP$gene, significant_genes_ea_wt_DN$gene)
ea_dependent_genes_list <- setdiff(ea_dependent_genes_list, significant_genes_ca_wt_DN$gene)
ea_dependent_genes_list <- intersect(ea_dependent_genes_list, significant_genes_ca_vs_ea$gene)

ordered_tbm_dependent_genes <- subset(res_bjab_ea_wt_list, gene %in% ea_dependent_genes_list) %>%
  dplyr::arrange(padj)

print(ordered_tbm_dependent_genes$gene)
```

```{r}
ca_ea_common_genes_list <- intersect(significant_genes_wt_mock_UP$gene, significant_genes_ea_wt_DN$gene)
ca_ea_common_genes_list <- intersect(ca_ea_common_genes_list, significant_genes_ca_wt_DN$gene)

ordered_codependent_genes <- subset(res_bjab_caea_wt_list, gene %in% ca_ea_common_genes_list) %>%
  dplyr::arrange(padj)

print(ordered_codependent_genes$gene)
```

```{r}
df <- as.data.frame(as.data.frame(coldata)$Sample_Group) 
rownames(df) <- row.names(coldata)
colnames(df) <- c("Sample_Group")
df <- df %>% dplyr::arrange(Sample_Group)

df$Sample_Group <- factor(df$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))

df <- df %>% arrange(Sample_Group)

df_subset <- subset(df, Sample_Group %in% c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))


# Find indices of matching column names in vsdMat
idx <- match(rownames(df_subset), colnames(vsdMat))

# Subset vsdMat using the matching indices
vsdMat_subset <- vsdMat[, idx]

data_heatmap <- subset(vsdMat_subset, rownames(vsdMat_subset) %in% c(ordered_codependent_genes$gene,
                                                                     ordered_protease_dependent_genes$gene, 
                                                                     ordered_tbm_dependent_genes$gene))
data_heatmap <- data_heatmap[order(match(rownames(data_heatmap),c(ordered_codependent_genes$gene,
                                                                     ordered_protease_dependent_genes$gene, 
                                                                     ordered_tbm_dependent_genes$gene))),,drop=F]


pdf("Heatmap_malt1kos.pdf", height = 7.5, width=7)
pheatmap(data_heatmap, 
         cluster_rows=FALSE, 
         show_rownames=TRUE, 
         annotation_col=df_subset, 
         scale = "row", 
         fontsize = 14, fontsize_col = 10, fontsize_row = 10, 
         cluster_cols = FALSE, 
         angle_col = 90, 
         gaps_row = c(length(ca_ea_common_genes_list), length(ca_ea_common_genes_list) + length(ca_dependent_genes_list)),
         annotation_colors = list(Sample_Group = c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                                                   "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                                                   "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                                                   "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                                                   "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998")),
         annotation_legend = FALSE)
dev.off()
```

```{r}
#These are the genes that are DN ONLY by the combination of CAmut+EAmut
# significant_genes_caea_wt_DN_ONLY <- setdiff(significant_genes_caea_wt_DN$gene, c(ca_ea_common_genes_list,ca_dependent_genes_list, ea_dependent_genes_list))
# 
# print(significant_genes_caea_wt_DN_ONLY)
```

```{r}
# significant_genes_t6ko_wt_mock_UP <- subset(res_t6ko_wt_mock_list, padj <= 0.1 & log2FoldChange > 0)  %>%
# dplyr::arrange(padj)
# 
# significant_genes_t6ko_ca_wt_DN <- subset(res_t6ko_ca_wt_list, padj <= 0.1 & log2FoldChange < 0)  %>%
# dplyr::arrange(padj)
# 
# #These are the genes that are UP by MALT1 and DN by CAmut when TRAF6 is KO
# t6ko_ca_dependent_genes_list <- intersect(significant_genes_t6ko_wt_mock_UP$gene, significant_genes_t6ko_ca_wt_DN$gene)
# print(t6ko_ca_dependent_genes_list)
# 
# t6ko_vs_eaca_genes_list <- setdiff(t6ko_ca_dependent_genes_list, significant_genes_caea_wt_DN_ONLY)
# print(t6ko_vs_eaca_genes_list)
```

```{r}
# df_subset_t6ko <- subset(df, Sample_Group %in% c( "BJAB_TRAF6_KO_MALT_KO_C11_L232LI_Mock",
#                                                                 "BJAB_TRAF6_KO_MALT_KO_C11_L232LI_WT",
#                                                                 "BJAB_TRAF6_KO_MALT_KO_C11_L232LI_CA"))
# 
# 
# # Find indices of matching column names in vsdMat
# idx <- match(rownames(df_subset_t6ko), colnames(vsdMat))
# 
# # Subset vsdMat using the matching indices
# vsdMat_subset_t6ko <- vsdMat[, idx]
# 
# data_heatmap <- subset(vsdMat_subset_t6ko, rownames(vsdMat_subset_t6ko) %in% significant_genes_t6ko_ca_wt_DN$gene)
# data_heatmap <- data_heatmap[order(match(rownames(data_heatmap),significant_genes_t6ko_ca_wt_DN$gene)),,drop=F]
# 
# 
# pdf("Heatmap_malt1kos_t6ko.pdf", height = 18, width=18)
# pheatmap(data_heatmap, cluster_rows=FALSE, show_rownames=TRUE, annotation_col=df_subset_t6ko, scale = "row", fontsize = 20, fontsize_col = 20, fontsize_row = 14, cluster_cols = FALSE, angle_col = 90)
# dev.off()
```

```{r}
# dds_counts <- dds@assays@data@listData[["counts"]]
# dds_counts_melted <- reshape2::melt(dds_counts)
# colnames(dds_counts_melted) <- c("gene", "Sample_Name", "Count")
# 
# dds_counts_melted_selectedgenes <- subset(dds_counts_melted, gene %in% head(ca_dependent_genes_list, n=24))
# 
# dds_counts_melted_selectedgenes_annotated <- merge(dds_counts_melted_selectedgenes, coldata, by="Sample_Name") 
# 
# dds_counts_melted_selectedgenes_annotated$Sample_Group <- factor(dds_counts_melted_selectedgenes_annotated$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
#                                                                 "BJAB_MALT_KO_C11_L232LI_WT",
#                                                                 "BJAB_MALT_KO_C11_L232LI_CA",
#                                                                 "BJAB_MALT_KO_C11_L232LI_EA",
#                                                                 "BJAB_MALT_KO_C11_L232LI_CA_EA"))
# 
# dds_counts_melted_selectedgenes_annotated <- dds_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)
# 
# dds_counts_melted_selectedgenes_annotated_subset <- subset(dds_counts_melted_selectedgenes_annotated, Sample_Group %in% c("BJAB_MALT_KO_C11_L232LI_Mock",
#                                                                 "BJAB_MALT_KO_C11_L232LI_WT",
#                                                                 "BJAB_MALT_KO_C11_L232LI_CA",
#                                                                 "BJAB_MALT_KO_C11_L232LI_EA",
#                                                                 "BJAB_MALT_KO_C11_L232LI_CA_EA"))
# 
# 
# pdf("Boxplots_ca_dependent.pdf", width = 6, height = 12)
# ggplot(dds_counts_melted_selectedgenes_annotated_subset, aes(x=Sample_Group, y=Count+1)) + 
#   geom_boxplot(outlier.shape = NA) +
#   geom_jitter(width = 0.1) +
#   facet_wrap2(~gene, axes = "all", remove_labels = "all", scales = "free_y", nrow = 6, ncol = 4) + # this can also just by "x" or "y" to remove axis labels
#   scale_y_log10() + 
#   theme_bw() +
#   theme(axis.title.x = element_blank(),
#         axis.ticks.x = element_blank(),
#         strip.background = element_blank(),
#         strip.placement = "outside",
#         strip.text = element_text(size = 10, face = "bold"),
#         axis.text.x = element_text(angle = 90))+
#   ylab("log10(Count+1)")
# dev.off()
```

```{r}
colnames(vsdmelted) <- c("gene", "Sample_Name", "Count")

vsd_counts_melted_selectedgenes <- subset(vsdmelted, gene %in% ca_dependent_genes_list)

vsd_counts_melted_selectedgenes_annotated <- merge(vsd_counts_melted_selectedgenes, coldata, by="Sample_Name") 

vsd_counts_melted_selectedgenes_annotated$Sample_Group <- factor(vsd_counts_melted_selectedgenes_annotated$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))

vsd_counts_melted_selectedgenes_annotated <- vsd_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)

vsd_counts_melted_selectedgenes_annotated_subset <- subset(vsd_counts_melted_selectedgenes_annotated, Sample_Group %in% c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))


pdf("Boxplots_ca_dependent.pdf", width = 6, height = 9)
ggplot(vsd_counts_melted_selectedgenes_annotated_subset, aes(x=Sample_Group, y=Count, color=Sample_Group)) + 
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(width = 0.1) +
  facet_wrap2(~gene, axes = "all", remove_labels = "all", scales = "free_y", nrow = 3, ncol = 3) + # this can also just by "x" or "y" to remove axis labels
  #scale_y_log10() + 
  theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(size = 10, face = "bold"),
        axis.text.x = element_text(angle = 90)) +
  ylab("Normalized Count") +
  scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                              "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                              "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                              "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                              "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998"))+
  guides(color=FALSE)
dev.off()


# Start a new PDF file with 1 plot per page
pdf("gene_boxplots_protease_dependent.pdf", width = 2.2, height=5)

# Loop over each gene and create a plot on a new page
for (g in unique(vsd_counts_melted_selectedgenes_annotated_subset$gene)) {
  # Subset the data for the current gene
  gene_data <- subset(vsd_counts_melted_selectedgenes_annotated_subset, gene == g)
  
  # Create the plot for the current gene
  p <- ggplot(gene_data, aes(x = Sample_Group, y = Count, color = Sample_Group)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(width = 0.1) +
    scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                                "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                                "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                                "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                                "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998")) +
    ylab("Normalized Count") +
    ggtitle(paste(g)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          plot.background = element_blank(), 
          axis.line = element_line(), 
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+
    # theme(axis.title.x = element_blank(),
    #       axis.ticks.x = element_blank(),
    #       strip.background = element_blank(),
    #       strip.placement = "outside",
    #       strip.text = element_text(size = 10, face = "bold"),
    #       axis.text.x = element_text(angle = 90))+ 
    guides(color=FALSE)
  
  # Print the plot for the current gene on a new page
  print(p)
}

# Close the PDF file
dev.off()
```

```{r}
colnames(vsdmelted) <- c("gene", "Sample_Name", "Count")

vsd_counts_melted_selectedgenes <- subset(vsdmelted, gene %in% ea_dependent_genes_list)

vsd_counts_melted_selectedgenes_annotated <- merge(vsd_counts_melted_selectedgenes, coldata, by="Sample_Name") 

vsd_counts_melted_selectedgenes_annotated$Sample_Group <- factor(vsd_counts_melted_selectedgenes_annotated$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))

vsd_counts_melted_selectedgenes_annotated <- vsd_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)

vsd_counts_melted_selectedgenes_annotated_subset <- subset(vsd_counts_melted_selectedgenes_annotated, Sample_Group %in% c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))


pdf("Boxplots_ea_dependent.pdf", width = 6, height = 6)
ggplot(vsd_counts_melted_selectedgenes_annotated_subset, aes(x=Sample_Group, y=Count, color = Sample_Group)) + 
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(width = 0.1) +
  facet_wrap2(~gene, axes = "all", remove_labels = "all", scales = "free_y", nrow = 2, ncol = 3) + # this can also just by "x" or "y" to remove axis labels
  scale_y_log10() + 
  theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(size = 10, face = "bold"),
        axis.text.x = element_text(angle = 90))  +
  ylab("Normalized Count") +
  scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                              "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                              "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                              "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                              "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998"))+
  guides(color=FALSE)
dev.off()

# Start a new PDF file with 1 plot per page
pdf("gene_boxplots_tbm_dependent.pdf", width = 2.2, height=5)

# Loop over each gene and create a plot on a new page
for (g in unique(vsd_counts_melted_selectedgenes_annotated_subset$gene)) {
  # Subset the data for the current gene
  gene_data <- subset(vsd_counts_melted_selectedgenes_annotated_subset, gene == g)
  
  # Create the plot for the current gene
  p <- ggplot(gene_data, aes(x = Sample_Group, y = Count, color = Sample_Group)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(width = 0.1) +
    scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                                "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                                "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                                "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                                "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998")) +
    ylab("Normalized Count") +
    ggtitle(paste(g)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          plot.background = element_blank(), 
          axis.line = element_line(), 
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ 
    guides(color=FALSE)
  
  # Print the plot for the current gene on a new page
  print(p)
}

# Close the PDF file
dev.off()
```

```{r}
colnames(vsdmelted) <- c("gene", "Sample_Name", "Count")

vsd_counts_melted_selectedgenes <- subset(vsdmelted, gene %in% ca_ea_common_genes_list)

vsd_counts_melted_selectedgenes_annotated <- merge(vsd_counts_melted_selectedgenes, coldata, by="Sample_Name") 

vsd_counts_melted_selectedgenes_annotated$Sample_Group <- factor(vsd_counts_melted_selectedgenes_annotated$Sample_Group, levels = c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))

vsd_counts_melted_selectedgenes_annotated <- vsd_counts_melted_selectedgenes_annotated %>% arrange(Sample_Group)

vsd_counts_melted_selectedgenes_annotated_subset <- subset(vsd_counts_melted_selectedgenes_annotated, Sample_Group %in% c("BJAB_MALT_KO_C11_L232LI_Mock",
                                                                "BJAB_MALT_KO_C11_L232LI_WT",
                                                                "BJAB_MALT_KO_C11_L232LI_CA",
                                                                "BJAB_MALT_KO_C11_L232LI_EA",
                                                                "BJAB_MALT_KO_C11_L232LI_CA_EA"))


pdf("Boxplots_ca_ea_dependent.pdf", width = 9, height = 12)
ggplot(vsd_counts_melted_selectedgenes_annotated_subset, aes(x=Sample_Group, y=Count)) + 
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(width = 0.1) +
  facet_wrap2(~gene, axes = "all", remove_labels = "all", scales = "free_y", nrow = 6, ncol = 6) + # this can also just by "x" or "y" to remove axis labels
  scale_y_log10() + 
  theme_bw() +
  theme(axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(size = 10, face = "bold"),
        axis.text.x = element_text(angle = 90)) 
dev.off()

# Start a new PDF file with 1 plot per page
pdf("gene_boxplots_codependent.pdf", width = 2.2, height=5)

# Loop over each gene and create a plot on a new page
for (g in unique(vsd_counts_melted_selectedgenes_annotated_subset$gene)) {
  # Subset the data for the current gene
  gene_data <- subset(vsd_counts_melted_selectedgenes_annotated_subset, gene == g)
  
  # Create the plot for the current gene
  p <- ggplot(gene_data, aes(x = Sample_Group, y = Count, color = Sample_Group)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(width = 0.1) +
    scale_color_manual(values=c("BJAB_MALT_KO_C11_L232LI_Mock" = "#1B9CD8",
                                "BJAB_MALT_KO_C11_L232LI_WT" = "#F39205",
                                "BJAB_MALT_KO_C11_L232LI_CA" = "#95C120",
                                "BJAB_MALT_KO_C11_L232LI_EA" = "#016633",
                                "BJAB_MALT_KO_C11_L232LI_CA_EA" = "#CA9998")) +
    ylab("Normalized Count") +
    ggtitle(paste(g)) +
    theme_bw() +
    theme(panel.grid.major = element_blank(), 
          panel.grid.minor = element_blank(),
          panel.background = element_blank(),
          plot.background = element_blank(), 
          axis.line = element_line(), 
          axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1))+ 
    guides(color=FALSE)
  
  # Print the plot for the current gene on a new page
  print(p)
}

# Close the PDF file
dev.off()
```


```{r}
a<-res_bjab_ea_wt_list %>% na.omit() %>% dplyr::select("log2FoldChange", "padj", "gene") %>% dplyr::mutate(minuslogp = -log10(padj), lfcsign = log2FoldChange/abs(log2FoldChange), signlogp = lfcsign*minuslogp) 
b<-res_bjab_ca_wt_list %>% na.omit() %>% dplyr::select("log2FoldChange", "padj", "gene")%>% dplyr::mutate(minuslogp = -log10(padj), lfcsign = log2FoldChange/abs(log2FoldChange), signlogp = lfcsign*minuslogp) 
c <- merge(a,b,by="gene") 


pdf(file="signlog_malt1mutants.pdf", width=6, height=6)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.3)+
  geom_point(data=subset(c, gene %in% c(ordered_codependent_genes$gene)), size=2.25,color="#CA9998", alpha=0.6)+
  geom_point(data=subset(c, gene %in% c(ordered_protease_dependent_genes$gene)), size=2.25,color="#95C120", alpha=0.6)+
  geom_point(data=subset(c, gene %in% c(ordered_tbm_dependent_genes$gene)), size=2.25,color="#016633", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  xlab("TBM vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  ylab("PM vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 12, face = "bold"), axis.text=element_text(size=10, face = "bold"))+
  geom_text_repel(data=subset(c, gene %in% c(ordered_codependent_genes$gene,
                                                  ordered_protease_dependent_genes$gene, 
                                                  ordered_tbm_dependent_genes$gene)), aes(label=gene), 
                  size=4, 
                  color="black", 
                  force=20,
                  min.segment.length = 0, 
                  max.overlaps = Inf)+
  scale_x_continuous(breaks = seq(-35,15,by=5))+
  scale_y_continuous(breaks = seq(-100,30,by=10))
dev.off()
```

